### DISCUSSION TITLE
Can this be used to perform CRUD inside our custom strapi plugin?

### DISCUSSION THREAD

**Discussion by IamSAL** [Stun3R/strapi-sdk-js#110] [Q&A]
Can this be used to perform CRUD inside our custom strapi plugin admin panel app, or is there a better way?

**Comment by Stun3R** [comment]
> Can this be used to perform CRUD inside our custom strapi plugin admin panel app, or is there a better way?

Hi @IamSAL 👋🏼 

The SDK follow the [REST API](https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/rest-api.html). In Strapi v4, you have access to the [Query Engine API](https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/query-engine-api.html), [Entity Service API](https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/entity-service-api.html) & Plugin APIs Reference so I think it would work to use the SDK if you trigger some REST endpoint BUT would be better to use the API I mentioned aboved.

**Reply by IamSAL** [reply]
Thank you so much for your response!

I am creating a strapi plugin that has two sides-admin and a server. I believe the Query engine and Entity service are only usable on the server side.
as the admin panel UI  is built upon react, is the Query Engine API is usable there also? if not, maybe your SDK can be the best fit. please guide and thanks :)

**Reply by Stun3R** [reply]
> Thank you so much for your response!
> 
> I am creating a strapi plugin that has two sides-admin and a server. I believe the Query engine and Entity service are only usable on the server side. as the admin panel UI is built upon react, is the Query Engine API is usable there also? if not, maybe your SDK can be the best fit. please guide and thanks :)

Maybe this would help for the admin side: https://docs.strapi.io/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.html. If not & if you want to trigger classic endpoint of your API then the SDK would suit perfectly
